topdown/eval.go

```diff
@@ -1250,9 +1255,9 @@ func (e *eval) getRules(ref ast.Ref) (*ast.IndexResult, error) {
 	var result *ast.IndexResult
 	var err error
 	if e.indexing {
-		result, err = index.Lookup(e)
+		result, err = index.Lookup(&evalResolver{e: e, args: args})
```

Comment:
:x: This isn't correct for (shallow) PE: we'll want all function bodies to be PE'ed, because any subsequent call will skip the partial eval. Hence the first occasion to PE a function shouldn't limit the evaluated function bodies.